### DISCUSSION TITLE
geojson layer vertical polygon

### DISCUSSION THREAD

**Discussion by radoslavirha** [visgl/deck.gl#6082] [Q&A]
Hi,

is it possible to display vertical polygons (window, billboard,...?) in deck.gl? I tried it, but polygon is not rendered. When I tilt it slightly, I can see my polygon. It's not the case with extruded polygons, it works perfectly.

My use case is to display different geometries with attributes and geojson seems to be the most suitable.

Thanks

**Comment by Pessimistress** [comment]
The GeoJsonLayer renders 3D polygons if the features' coordinates have 3 numbers.

Do you mean that you cannot see a vertical polygon from top- down (pitch: 0)? That's expected because it does not have any volume. The GeoJsonLayer (which renders SolidPolygonLayer) only extrudes polygons along the z axis. If you want to extrude a vertical box, either draw the thin base and set `getElevation` to the vertical height, or use the SimpleMeshLayer for more complex shapes.

**Reply by radoslavirha** [reply]
Yes I want polygons with 0 pitch. Extruded version should work, but from my point of view it's not suitable. I'm talking about arbitrary polygon geometries (mostly polygons), so it'll create very specific exception only for vertical polygons.

SimpleMeshLayer looks fine, but it seems I need to create separate layers for different geometries (4x2 m polygon, 2x3m polygon,....).

Is it hard to prototype SolidPolygonLayer to create volume for vertical polygons? I prefer GeoJSON over meshes, it's easier to store it in database, work with attributes,...

**Reply by ibgreen** [reply]
It might be worth clarifying how the current polygon tesselation algorithm (earcut) works with polygons that are not parallel to the x,y-plane. The documentation doesn't seem to cover it though I saw some mention of z-coordinate improvements. 

If it earcut does handle this, it may be something minor we do that gets in the way. 

If it doesn't, arbitrarily oriented polygons would likely need to be transformed (rotated) into x,y coordinates before being tessellated and then generated triangles would need to be rotated back. 

This might be doable if your linear algebra is strong, but probably it would require a few days of somewhat fiddly work to get everything sorted.

**Reply by radoslavirha** [reply]
Where can I find something about tesselation algorithm, or what it uses under the hood?

**Reply by Pessimistress** [reply]
> the tesselation algorithm

[earcut](https://www.npmjs.com/package/earcut)
